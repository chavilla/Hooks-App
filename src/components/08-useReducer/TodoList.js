import React from "react";
import PropTypes from 'prop-types';
import TodoItem from "./TodoItem";

const TodoList = ({ todo, i,handleDelete, handleToggle }) => {
  return (
      <TodoItem
      key={todo.id }
      i={i}
      todo={todo}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
      />
  );
};


TodoList.propTypes={
  todo: PropTypes.object.isRequired,
  i:PropTypes.number.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleToggle:PropTypes.func.isRequired
}



export default TodoList;
