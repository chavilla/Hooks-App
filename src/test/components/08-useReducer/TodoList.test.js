import { shallow } from 'enzyme';
import React from 'react';
import { state } from '../../fixtures/toDoList';
import TodoList from '../../../components/08-useReducer/TodoList';

describe('TodoList Testing', () => {

    const handleToggle=jest.fn();
    const handleDelete=jest.fn();

    const wrapper=shallow(<TodoList 
    handleToggle={handleToggle}
    handleDelete={handleDelete}
    todo={state}
    i={0}
    />);

    test('should to match snapshot', () => {
      
        expect(wrapper).toMatchSnapshot();

    });

    test('should return the size of the array', () => {
      
        console.log(wrapper.html()); return;

        //expect(wrapper.find('TodoItem').length).toBe(2);

    }); 
});