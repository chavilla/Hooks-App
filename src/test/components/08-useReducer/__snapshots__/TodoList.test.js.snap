// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Testing should to match snapshot 1`] = `
<TodoItem
  handleDelete={[MockFunction]}
  handleToggle={[MockFunction]}
  i={0}
  key="1"
  todo={
    Object {
      "done": false,
      "id": 1,
      "name": "Aprender InglÃ©s",
    }
  }
/>
`;
